js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetSelectionSummaryTile.tsx

```diff
@@ -10,39 +11,54 @@ import {AssetHealthStatus} from '../../graphql/types';
 import {numberFormatter} from '../../ui/formatters';
 import {AssetTableFragment} from '../types/AssetTableFragment.types';
 import {useAllAssets} from '../useAllAssets';
+import {ViewType} from './util';
 
 export const TILE_WIDTH = 230;
 export const TILE_HEIGHT = 108;
 export const TILE_GAP = 8;
 
+// An in memory cache to side step slow asset selection filtering when revisiting the page.
+// To fix this properly we need to add more caches within useAssetSelectionFiltering and useAssetGraphData but it is difficult to do so
+// since the array of nodes they receive aren't the same when you visit the page again since they're the result of `.filter` calls.
+const memoryCache = new Map<string, {assets: any[]}>();
+
 export const AssetSelectionSummaryTileFromSelection = React.memo(
   ({
     icon,
-    label,
     selection,
-    link,
   }: {
     icon: React.ReactNode;
-    label: string;
-    selection: string;
-    link: string;
+    selection: Extract<ViewType, {__typename: 'CatalogView'}>;
   }) => {
-    const {assets, loading} = useAllAssets();
-    const {filtered} = useAssetSelectionFiltering({
-      assets,
-      assetSelection: selection,
-      loading,
+    const assetSelection = selection.selection.querySelection ?? '';
+    const {assets: allAssets, loading: allAssetsLoading} = useAllAssets();
+
+    const {filtered, loading: filteredLoading} = useAssetSelectionFiltering({
```

Comment:
This all looks good but longer term I think we should investigate why this is slow rather than trying to memoize + cache. I'd expect this to run in 100ms even on 200k assets, since it's just a pile of .filter calls? Maybe there's some graph traversal or weird string stuff in here that is slowing it and we could memoize down inside those specific filters?

Reply:
It is pretty fast, but there are multiple instances of this component on the page so I want to memoize as much as possible to avoid a long frame.